tests/test_sentinel_managed_connection.py

```diff
@@ -0,0 +1,209 @@
+from typing import Tuple
+from unittest import mock
+
+import pytest
+from redis import Redis
+from redis.sentinel import Sentinel, SentinelConnectionPool, SentinelManagedConnection
+from redis.utils import HIREDIS_AVAILABLE
+
+pytestmark = pytest.mark.skipif(HIREDIS_AVAILABLE, reason="PythonParser only")
```

Comment:
I marked these test to ignore Hiredis. I'm not very familiar with Hiredis setup, but it looks like the hiredis' code paths doesn't really work well when we call connection-related operations like `can_read` (and as far as I understand it is unrelated to my change). Let me know if i need to not skip Hiredis